server/src/test/java/org/candlepin/dto/api/v1/ActivationKeyDTOTest.java

```diff
@@ -0,0 +1,422 @@
+        super(ActivationKeyDTO.class);
+
+        OwnerDTO owner = new OwnerDTO();
+        owner.setId("owner_id");
+        owner.setKey("owner_key");
+        owner.setDisplayName("owner_name");
+        owner.setContentPrefix("content_prefix");
+        owner.setDefaultServiceLevel("service_level");
+        owner.setLogLevel("log_level");
+        owner.setAutobindDisabled(true);
+        owner.setContentAccessMode("content_access_mode");
+        owner.setContentAccessModeList("content_access_mode_list");
+
+        this.values = new HashMap<String, Object>();
+        this.values.put("Id", "test-id");
+        this.values.put("Name", "test-name");
+        this.values.put("Description", "test-description");
+        this.values.put("Owner", owner);
+        this.values.put("ReleaseVersion", "test-release-ver");
+        this.values.put("ServiceLevel", "test-service-level");
+        this.values.put("AutoAttach", true);
+        this.values.put("Created", new Date());
+        this.values.put("Updated", new Date());
+
+        Set<ActivationKeyDTO.ActivationKeyPoolDTO> pools =
+            new HashSet<ActivationKeyDTO.ActivationKeyPoolDTO>();
+        pools.add(new ActivationKeyDTO.ActivationKeyPoolDTO("test-id-pool", 1L));
+        this.values.put("Pools", pools);
+
+        Set<String> prods =
+            new HashSet<String>();
+        prods.add("test-id-prod");
+        this.values.put("Products", prods);
+
+        Set<Map<String, String>> productDTOs = new HashSet<Map<String, String>>();
+        Map<String, String> productDTO = new HashMap<String, String>();
+        productDTO.put("productId", "test-id-prodDto");
+        productDTOs.add(productDTO);
+        this.values.put("ProductDTOs", productDTOs);
+
+        Set<ActivationKeyDTO.ActivationKeyContentOverrideDTO> overrides =
+            new HashSet<ActivationKeyDTO.ActivationKeyContentOverrideDTO>();
+        overrides.add(new ActivationKeyDTO.ActivationKeyContentOverrideDTO(
+            "test-contentLabel-override",
+            "test-name-override",
+            "test-value-override"));
+        this.values.put("ContentOverrides", overrides);
+    }
+
+    /**
+     * @{inheritDocs}
+     */
+    @Override
+    protected Object getInputValueForMutator(String field) {
+        return this.values.get(field);
+    }
+
+    /**
+     * @{inheritDocs}
+     */
+    @Override
+    protected Object getOutputValueForAccessor(String field, Object input) {
+        // Nothing to do here
+        return input;
+    }
+
+    @Test
+    public void testHasProductWithAbsentProduct() {
+        ActivationKeyDTO dto = new ActivationKeyDTO();
+        Product prod = new Product();
+        prod.setId("test-id-prod-1");
+        assertFalse(dto.hasProduct(prod));
+    }
+
+    @Test
+    public void testHasProductWithPresentProduct() {
+        ActivationKeyDTO dto = new ActivationKeyDTO();
+        Set<String> products =
+            new HashSet<String>();
```

Comment:
Same here. Try to avoid wrapping if you can. To avoid littering this review with more of those comments, this is the last one I'll call out. Give the whole branch a look for more of these to avoid.